plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToJobNodeConverter.java

```diff
@@ -136,17 +145,36 @@ public IRepositoryFileData convert( final InputStream inputStream, final String
       Document doc = PDIImportUtil.loadXMLFrom( inputStream );
       if ( doc != null ) {
         jobMeta.loadXML( doc.getDocumentElement(), repository, null );
+        if ( jobMeta.hasMissingPlugins() ) {
+          KettleMissingPluginsException
+            missingPluginsException =
+            new KettleMissingPluginsException( getErrorMessage( jobMeta.getMissingEntries() ) );
+          throw new ConverterException( missingPluginsException );
+        }
         JobDelegate delegate = new JobDelegate( repository, this.unifiedRepository );
         delegate.saveSharedObjects( jobMeta, null );
         return new NodeRepositoryFileData( delegate.elementToDataNode( jobMeta ), size );
       } else {
         return null;
       }
-    } catch ( Exception e ) {
+    } catch ( IOException | KettleException e ) {
       return null;
     }
   }
 
+  private String getErrorMessage( List<MissingEntry> missingEntries ) {
+
+    StringBuilder entries = new StringBuilder();
+    for ( MissingEntry entry : missingEntries ) {
+      if ( missingEntries.indexOf( entry ) == missingEntries.size() - 1 ) {
```

Comment:
Shouldn't this be refactored to something like this?

entries.append( "- " + entry.getName() + " - " + entry.getMissingPluginId() + "\n");
if ( missingEntries.indexOf( entry ) == missingEntries.size() - 1 ) {
  entries.append("\n");
}

Reply:
I could do that. Nevertheless, performance wise is the same thing. Since we are making one if evaluation at each iteration.

Reply:
Done